src/ui/common/src/components/pages/metric/id/index.tsx

```diff
@@ -0,0 +1,462 @@
+      nodeType: NodeType.GenericArtifact,
+      nodeIcon: genericArtifactNodeIcon,
+    },
+    {
+      name: 'image_artifact_node',
+      nodeType: NodeType.ImageArtifact,
+      nodeIcon: imageArtifactNodeIcon,
+    },
+    {
+      name: 'jsonArtifactNode',
+      nodeType: NodeType.JsonArtifact,
+      nodeIcon: jsonArtifactNodeIcon,
+    },
+    {
+      name: 'metric_operator_node',
+      nodeType: NodeType.MetricOp,
+      nodeIcon: metricOperatorNodeIcon,
+    },
+    {
+      name: 'numeric_artifact_node',
+      nodeType: NodeType.NumericArtifact,
+      nodeIcon: numericArtifactNodeIcon,
+    },
+    {
+      name: 'string_artifact_node',
+      nodeType: NodeType.StringArtifact,
+      nodeIcon: stringArtifactNodeIcon,
+    },
+    {
+      name: 'table_artifact_node_icon',
+      nodeType: NodeType.TableArtifact,
+      nodeIcon: tableArtifactNodeIcon,
+    },
+  ];
+
+  const artifactTypeToIconMapping = {
+    [ArtifactType.String]: stringArtifactNodeIcon,
+    [ArtifactType.Bool]: boolArtifactNodeIcon,
+    [ArtifactType.Numeric]: numericArtifactNodeIcon,
+    [ArtifactType.Dict]: dictArtifactNodeIcon,
+    // TODO: figure out if we should use other icon for tuple
+    [ArtifactType.Tuple]: dictArtifactNodeIcon,
+    [ArtifactType.Table]: tableArtifactNodeIcon,
+    [ArtifactType.Json]: jsonArtifactNodeIcon,
+    // TODO: figure out what to show for bytes.
+    [ArtifactType.Bytes]: dictArtifactNodeIcon,
+    [ArtifactType.Image]: imageArtifactNodeIcon,
+    // TODO: Figure out what to show for Picklable
+    [ArtifactType.Picklable]: dictArtifactNodeIcon,
+  };
+
+  const listStyle = {
+    width: '100%',
+    maxWidth: 360,
+    bgcolor: 'background.paper',
+  };
+
+  // return null if we don't have the workflow loaded.
+  // This workflow doesn't exist.
+  if (workflow.loadingStatus.loading === LoadingStatusEnum.Failed) {
+    navigate('/404');
+    return null;
+  }
+
+  if (operator) {
+    const inputs = operator.inputs;
+    const outputs = operator.outputs;
+
+    const operatorArtifacts = [...inputs, ...outputs];
+    // fetch output artifacts.
+    operatorArtifacts.map((artifactId) => {
+      if (!workflow.artifactResults[artifactId])
+        dispatch(
+          handleGetArtifactResults({
+            apiKey: user.apiKey,
+            workflowDagResultId,
+            artifactId,
+          })
+        );
+    });
+  }
+
+  if (!metricResult || !metricResult.result) {
+    return (
+      <Layout user={user}>
+        <CircularProgress />
+      </Layout>
+    );
+  }
+
+  // Function to get the numerical value of the metric output
+  const getOperatorOutput = () => {
+    if (!operator || !operator.outputs) {
+      return <CircularProgress />;
+    }
+
+    return operator.outputs.map((artifactId) => {
+      const artifactResult = workflow.artifactResults[artifactId];
+      if (!artifactResult) {
+        return null;
+      }
+
+      // TODO: Check the serialization_type of the artifacts and show a link
```

Comment:
@likawind we can get this from the spec right?

Reply:
we don't have that for artifact yet but it's easy to add. I didn't include serialization_type in first cut just because I didn't think we are using it

Reply:
and yes, assuming we have it we should get it from output artifact, which is easy to figure out using `.outputs` fields of this operator